bundle/edu.gemini.catalog/src/main/scala/edu/gemini/catalog/api/MagnitudeConstraints.scala

```diff
@@ -45,36 +47,57 @@ object SaturationConstraint {
     Order.orderBy(_.brightness)
 }
 
+sealed trait MagnitudeFilter {
+  def filter(t: SiderealTarget): Boolean
+
+  def referenceBand: MagnitudeBand
+  def faintnessConstraint: FaintnessConstraint
+  def saturationConstraint: Option[SaturationConstraint]
+}
+
 /**
- * Describes constraints for the magnitude of a target
- * See OT-19.
+ * Defines a typeclass of classes that can adjust MagnitudeConstraints, e.g. given certain conditions
  */
-case class MagnitudeConstraints(band: MagnitudeBand, faintnessConstraint: FaintnessConstraint, saturationConstraint: Option[SaturationConstraint]) {
+trait ConstraintsAdjuster[T] {
+  def adjust(t: T, mc: MagnitudeConstraints): MagnitudeConstraints
+}
 
-  def this(faint:Magnitude) = this(faint.band, FaintnessConstraint(faint.value), None)
+/**
+ * Defines a class that can extract a magnitude from a target
+ */
+sealed trait MagnitudeExtractor {
+  def extract(t: SiderealTarget):Option[Magnitude]
+  def bandSupported(b: MagnitudeBand):Boolean
+}
 
-  /**
-   * Maps a transformation into a new MagnitudeConstraints
-   */
-  def map(f: Magnitude => Magnitude): MagnitudeConstraints = {
-    val mappedFaintness = f(faintnessConstraint.toMagnitude(band)) 
-    val mappedSaturation = saturationConstraint.map(_.toMagnitude(band)).map(f)
+/**
+ * Search for a specific band on the target
+ */
+case class SingleBandExtractor(band: MagnitudeBand) extends MagnitudeExtractor {
+  override def extract(t: SiderealTarget) = t.magnitudeIn(band)
+  override def bandSupported(b: MagnitudeBand) = b === band
+}
 
-    val fl = FaintnessConstraint(mappedFaintness.value)
-    val sl = mappedSaturation.map(s => SaturationConstraint(s.value))
+/**
+ * Finds the first on a list of bands
+ */
+case class FirstBandExtractor(bands: NonEmptyList[MagnitudeBand]) extends MagnitudeExtractor {
+  override def extract(t: SiderealTarget) = bands.map(t.magnitudeIn).list.find(_.isDefined).flatten
+  override def bandSupported(b: MagnitudeBand) = bands.list.contains(b)
+}
```

Comment:
I sort of regret the "r like" band idea.  We might have just left everything defined in terms of a single band and just run the search (on the same candidates) 3 times?  At any rate, now that we have it, maybe we should push this idea up so that we aren't passing around single bands but rather something like this extractor idea that groups and handles mapping to the "contained" if you will bands.  It's just seems too weird to ask for `R` and get `UC`, etc.